src/ngx_http_lua_kong_module.c

```diff
@@ -325,23 +325,31 @@ ngx_http_lua_kong_set_upstream_ssl(ngx_http_request_t *r, ngx_connection_t *c)
     STACK_OF(X509)              *chain;
     EVP_PKEY                    *pkey;
     X509                        *x509;
+    X509_STORE                  *store;
 #ifdef OPENSSL_IS_BORINGSSL
     size_t                       i;
 #else
     int                          i;
 #endif
 
     ctx = ngx_http_get_module_ctx(r, ngx_http_lua_kong_module);
-    if (ctx == NULL || ctx->upstream_client_certificate_chain == NULL) {
+
+    if (ctx == NULL || (ctx->upstream_client_certificate_chain == NULL &&
```

Comment:
Since now we can set either client cert or verify param, this check can be changed to check for `ctx == NULL` only, and use `if` branch below to distinguish between those.